signer/tests/integration/postgres.rs

```diff
@@ -5142,6 +5070,39 @@ async fn verification_status_one_way_street() {
     signer::testing::storage::drop_db(db).await;
 }
 
+/// Return a stacks block that is anchored to the bitcoin block with the
+/// given block hash that is also on the canonical stacks blockchain, if
+/// one exists. If one doesn't exist then return the stacks chain tip.
+async fn get_stacks_block(
+    test_data: &TestData,
+    db: &PgStore,
+    anchor: &BitcoinBlockHash,
+) -> StacksBlockHash {
+    let request_stacks_blocks = test_data
+        .stacks_blocks
+        .iter()
+        .filter(|block| &block.bitcoin_anchor == anchor);
+
+    let (_, stacks_chain_tip) = db.get_chain_tips().await;
+
+    for block in request_stacks_blocks {
+        let in_canonical_stacks_blockchain = db
+            .in_canonical_stacks_blockchain(
+                &stacks_chain_tip,
+                &block.block_hash,
+                block.block_height,
+            )
+            .await
+            .unwrap();
+
+        if in_canonical_stacks_blockchain {
+            return block.block_hash;
+        }
+    }
+
+    stacks_chain_tip
```

Comment:
qq: I'm wondering if returning `stacks_chain_tip` if no block is found could be an "unexpected" result, vs just panicking. If your test setup was correct you shouldn't (I think?) get here.

Reply:
Yeah, I wasn't sure if panicking was the right thing to do, but having thought about this after a brief hiatus, it's better to panic since you probably expect there to be a Stacks block for the given bitcoin block. And if not you may need something other than the chain tip anyway. Updated in https://github.com/stacks-sbtc/sbtc/pull/1559/commits/1c849153462c792b20b2d0c1dea4ec7e62412baf.